test/end-to-end/offer.js

```diff
@@ -628,6 +629,34 @@ describe("Offer endpoint tests", () => {
                 expect(res.status).toBe(HTTPStatus.OK);
             });
         });
+
+        describe("Incomplete registration of the offer's company", () => {
+            beforeAll(async () => {
+                await Company.findOneAndUpdate({ name: test_company.name }, { hasFinishedRegistration: false });
+            });
+
+            afterAll(async () => {
+                await Company.findOneAndUpdate({ name: test_company.name }, { hasFinishedRegistration: true });
+            });
```

Comment:
You should not reuse the same company here, what would happen if some other test (from a sibling describe) were to run at the same time? Tests cannot depend/affect one another.